core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -78,7 +77,7 @@ export class OptimizedSsrEngine {
         }
       : undefined;
 
-    this.logger = this.ssrOptions?.logger as ExpressLoggerService; // we are sure the logger is defined in this place
+    this.logger = this.ssrOptions?.logger as ExpressServerLogger; // we are sure the logger is defined in this place
```

Comment:
```suggestion
    if (!this.ssrOptions?.logger) {
      throw new Error('`SsrOptimizationOptions.logger` is not defined');
    }
    this.logger = this.ssrOptions?.logger as ExpressServerLogger;
```

Reply:
We're pretty sure it's defined as if it's not provided by customers, it takes default value from default optimization options which (probably) cannot be overwritten by customers.

But I agree - can add this extra step for runtime code safety.

Reply:
I believe `as ExpressServerLogger;` is no longer needed as we checked if the value is defined.